test/system/500-networking.bats

```diff
@@ -925,13 +947,16 @@ EOF
 
     # Check with ns:/PATH
     if ! is_rootless; then
-        netns=netns$(random_string)
+        netns=netns-$(safename)
+        echo "$_LOG_PROMPT ip netns add $netns"
         ip netns add $netns
         run_podman create --network=ns:/var/run/netns/$netns $IMAGE
         cid=${output}
         run_podman inspect --format '{{ .NetworkSettings.Networks }}' $cid
         is "$output" 'map[]' "NeworkSettings should be empty"
         run_podman rm $cid
+        echo "$_LOG_PROMPT ip netns delete $netns"
+        ip netns delete $netns
```

Comment:
This fixes a leak. Found by a leak check I've added (experimental) to #23275.

Reply:
Not blocking but doesn't this mean this always leaks the netns when one of the checks above fails?
Should this be moved into a custom teardown function?

Reply:
Maybe, but what's different about here vs all the other places that leak netns on failure? Like [this one](https://api.cirrus-ci.com/v1/artifact/task/4687505503551488/html/sys-podman-fedora-39-root-host-boltdb.log.html#t--00369p) (scroll to bottom): every time `600 shell completion` fails, it leaks four netns, the UUID-named ones, even though the `network create` in that test uses `safename`. It has been bothering me for a while, but I've only spent minimal time investigating. 

Anyhow, if netns cleanup is more important than other container-related cleanup, I'm happy to move that into teardown somehow.

Reply:
if the netns files created by podman are leaked then this is a podman bug, even in the error paths we should be removing them

I don't think it is super important as it would only leak on error cases and there is no problems with leaks other than a bit of kernel memory keeping it around. And a reboot fixes that so I think that is fine for machines where these tests are run.